docs/design/contextual-tabs.md

```diff
@@ -0,0 +1,392 @@
+          "icon": [
+            {
+                "size": 16,
+                "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/Group16x16.png"
+            },
+            {
+                "size": 32,
+                "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/Group32x32.png"
+            },
+            {
+                "size": 80,
+                "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/Group80x80.png"
+            }
+          ],
+          "controls": [
+            {
+                "type": "Button",
+                "id": "CtxBt112",
+                "actionId": "executeWriteData",
+                "enabled": false,
+                "label": "Write Data",
+                "superTip": {
+                    "title": "Data Insertion",
+                    "description": "Use this button to insert data into the document."
+                },
+                "icon": [
+                    {
+                        "size": 16,
+                        "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/WriteDataButton16x16.png"
+                    },
+                    {
+                        "size": 32,
+                        "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/WriteDataButton32x32.png"
+                    },
+                    {
+                        "size": 80,
+                        "sourceLocation": "https://cdn.contoso.com/addins/datainsertion/Images/WriteDataButton80x80.png"
+                    }
+                ]
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}'
+```
+
+## Register the contextual tab with Office with requestCreateControls
+
+The contextual tab is registered with Office by calling the [Office.ribbon.requestCreateControls](/javascript/api/office/office.ribbon?view=common-js&preserve-view=true#requestcreateconrtols-input-) method in either the function that is assigned to `Office.initialize` or with the `Office.onReady` method. For more about these methods and initializing the add-in, see [Initialize your Office Add-in](../develop/initialize-add-in.md).
+
+The following is an example. Note that the JSON string must be converted to a JavaScript object with the `JSON.parse` method before it can be passed to a JavaScript function.
+
+```javascript
+Office.onReady(async () => {
+    const contextualTabJSON = ' ... ' // Assign the JSON string such as the one at the end of the preceding section.
+    const contextualTab = JSON.parse(contextualTabJSON);
+    await Office.ribbon.requestCreateControls(contextualTab);
+});
+```
+
+## Specify the contexts when the tab will be visible with requestUpdate
+
+Typically, a contextual tab should appear when a user-initiated event changes the add-in context. Consider a scenario in which the tab should be visible when, and only when, a chart (on the default worksheet of an Excel workbook) is activated.
```

Comment:
This is the behavior of built-in Contextual Tabs and you can do the same with Custom Contextual Tabs.